jinahub/encoders/image/TimmImageEncoder/tests/unit/test_unit.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from typing import Dict, Tuple
+
+import numpy as np
+import pytest
+from jina import Document, DocumentArray, Executor
+
+from ...timm_encoder import TimmImageEncoder
+
+
+def test_config():
+    ex = Executor.load_config(str(Path(__file__).parents[2] / "config.yml"))
+    assert ex.default_batch_size == 32
+
+
+@pytest.mark.parametrize(
+    ["content", "out_shape"],
+    [
+        ([np.ones((10, 10, 3), dtype=np.uint8), (3, 224, 224)]),
+        ([np.ones((360, 420, 3), dtype=np.uint8), (3, 224, 224)]),
+        ([np.ones((300, 300, 3), dtype=np.uint8), (3, 224, 224)]),
+    ],
+)
+def test_preprocessing_reshape_correct(content: np.ndarray, out_shape: Tuple):
+    encoder = TimmImageEncoder()
+
+    reshaped_content = encoder._preprocess(content)
+
+    assert (
+        reshaped_content.shape == out_shape
+    ), f"Expected shape {out_shape} but got {reshaped_content.shape}"
+
+
+@pytest.mark.parametrize(
+    "traversal_paths, docs",
+    [
+        (("r",), pytest.lazy_fixture("docs_with_blobs")),
+        (("c",), pytest.lazy_fixture("docs_with_chunk_blobs")),
+    ],
+)
+def test_encode_image_returns_correct_length(
+    traversal_paths: Tuple[str], docs: DocumentArray
+) -> None:
+    encoder = TimmImageEncoder(default_traversal_path=traversal_paths)
+
+    encoder.encode(docs=docs, parameters={})
+
+    for doc in docs.traverse_flat(traversal_paths):
+        assert doc.embedding is not None
+        assert doc.embedding.shape == (512,)
+
+
+@pytest.mark.parametrize(
+    "model_name", ["resnet50", "mobilenetv3_large_100", "efficientnet_b1"]
+)
+def test_encodes_semantic_meaning(test_images: Dict[str, np.array], model_name: str):
+    encoder = TimmImageEncoder(model_name=model_name)
+    embeddings = {}
+
+    for name, image_arr in test_images.items():
+        docs = DocumentArray([Document(blob=image_arr)])
+        encoder.encode(docs, parameters={})
+        embeddings[name] = docs[0].embedding
+
+    def dist(a, b):
+        a_embedding = embeddings[a]
+        b_embedding = embeddings[b]
+        return np.linalg.norm(a_embedding - b_embedding)
```

Comment:
Can you rework this test to use `DocumentArray.match` method? See an example in `CLIPImageEncoder`